src/axom/slic/streams/GenericOutputStream.cpp

```diff
@@ -80,10 +104,38 @@ void GenericOutputStream::append(message::Level msgLevel,
 }
 
 //------------------------------------------------------------------------------
-void GenericOutputStream::outputLocal() { m_stream->flush(); }
+void GenericOutputStream::openBeforeFlush()
+{
+  if(m_isOstreamOwnedBySLIC && !m_opened)
+  {
+    std::ostringstream* oss = dynamic_cast<std::ostringstream*>(m_stream);
+    if(oss != nullptr)
+    {
+      std::string buffer = oss->str();
+      if(!buffer.empty())
+      {
+        delete m_stream;
+        m_stream = new std::ofstream(m_file_name);
+        (*m_stream) << buffer;
+        m_opened = true;
+      }
+    }
```

Comment:
Interesting approach! 

It took me a moment to figure out why the delete/new is necessary. 

Could you please add a brief comment that we're converting from the original `ostringstream` to an `ofstream` and writing the former's contents to the latter?